octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -73,12 +81,144 @@ async def initialize_impl(self):
         except ccxt.AuthenticationError:
             raise ccxt.AuthenticationError
 
+    async def parse_order(self, raw_order: dict, order_type: str = None,
+                          quantity: decimal.Decimal = None, price: decimal.Decimal = None,
+                          status: str = None, symbol: str = None,
+                          side: str = None, timestamp: int or float = None,
+                          check_completeness: bool = True) -> dict:
+        """
+        use this method to parse a single order
+
+        :param raw_order:
+
+        optional:
+        :param status: to use if it's missing in the order
+        :param order_type: to use if it's missing in the order
+        :param price: to use if it's missing in the order
+        :param quantity: to use if it's missing in the order
+        :param symbol: to use if it's missing in the order
+        :param side: to use if it's missing in the order
+        :param timestamp: to use if it's missing in the order
+
+        :param check_completeness: if true checks all attributes, 
+            if somethings missing it'll try to fetch it from the exchange
+
+        :return: formatted order dict (100% complete or we raise NotImplemented)
+        """
+        _parser = self.connector_config.ORDERS_PARSER(self.exchange_manager.exchange)
+        return await _parser.parse_order(raw_order, order_type=order_type, quantity=quantity,
+                                         price=price, status=status, symbol=symbol, side=side,
+                                         timestamp=timestamp, check_completeness=check_completeness)
+
+    async def parse_orders(self, raw_orders: list, order_type: str = None,
+                           quantity: decimal.Decimal = None, price: decimal.Decimal = None,
+                           status: str = None, symbol: str = None,
+                           side: str = None, timestamp: int or float = None,
+                           check_completeness: bool = True) -> list:
+        """
+        use this method to format a list of order dicts
+
+        :param raw_orders: raw orders with eventually missing data
+
+        optional:
+        :param status: to use if it's missing in the order
+        :param order_type: to use if it's missing in the order
+        :param price: to use if it's missing in the order
+        :param quantity: to use if it's missing in the order
+        :param symbol: to use if it's missing in the order
+        :param side: to use if it's missing in the order
+        :param timestamp: to use if it's missing in the order
+
+        :param check_completeness: if true checks all attributes, 
+            if somethings missing it'll try to fetch it from the exchange
+
+        :return: formatted orders list (100% complete or we raise NotImplemented report)
+            
+        """
+        _parser = self.connector_config.ORDERS_PARSER(self.exchange_manager.exchange)
```

Comment:
It's a generic implementation, it think we should put this in spot_exchange as it's not ccxt anymore with the unified parser

Reply:
done